src/melee.cpp

```diff
@@ -344,9 +345,17 @@ void player::melee_attack(Creature &t, bool allow_special, const matec_id &force
 
         // Make a rather quiet sound, to alert any nearby monsters
         if (!is_quiet()) { // check martial arts silence
-            sounds::sound( pos3(), 8, "", false, "melee_hit", weapon.typeId() );
+            sounds::sound( pos3(), 8, "", false, "", "" );
+        }
+        std::string material = "flesh";
+        bool targ_mon = (!t.is_npc() && !t.is_player());
+        if ( targ_mon ) {
```

Comment:
Just `if( t.is_monster() ) ` is fine.

----

I see this as even better:
```C++
std::string material = "flesh";
const monster *m = dynamic_cast<const monster*>( &t ); 
if( m != nullptr ) {
```

Just in case the `is_monster`  functions lies and `&t` can not be casted to a `monster*`.